sky/utils/control_master_checks.py

```diff
@@ -0,0 +1,49 @@
+"""Utils to check if the ssh control master should be disabled."""
+
+from functools import lru_cache
+
+from sky import sky_logging
+from sky.utils import subprocess_utils
+
+logger = sky_logging.init_logger(__name__)
+
+
+def is_tmp_9p_filesystem() -> bool:
+    """Check if the /tmp filesystem is 9p.
+
+    Returns:
+        bool: True if the /tmp filesystem is 9p, False otherwise.
+    """
+
+    result = subprocess_utils.run(['df', '-T', '/tmp'],
+                                  capture_output=True,
+                                  text=True,
+                                  shell=None,
+                                  check=None,
+                                  executable=None)
+
+    if result.returncode != 0:
+        return False
+
+    filesystem_infos = result.stdout.strip().split('\n')
+    if len(filesystem_infos) < 2:
+        return False
+    filesystem_types = filesystem_infos[1].split()
+    if len(filesystem_types) < 2:
+        return False
+    return filesystem_types[1].lower() == '9p'
+
+
+@lru_cache
```

Comment:
style: import module instead of functions
```suggestion
@functools.lru_cache
```